python_modules/dagster/dagster/_core/instance/ref.py

```diff
@@ -279,6 +279,9 @@ def config_defaults(base_dir):
                 "DefaultRunLauncher",
                 yaml.dump({}),
             ),
+            # For back-compat, the default is actually set in the secrets_loader property above,
+            # so that old clients loading new config don't try to load a class that they
+            # don't recognize
```

Comment:
when exactly do we hit this condition of old trying to load via this config? Is it just the release or 2 that we had the `secrets_loader_data` field on `InstanceRef` but the `EnvFileLoader` doesn't exist?

Reply:
Yeah this is unfortunate planning on my part. If I had introduced this last week this in the same release that we added this secrets_loader field (like the original version of my PR even did!) then this wouldn't be a problem

Reply:
can we add a test for this specific thing? just to encode this known issue in the code base. Truly emulating in the test would be a lot but poking at a serialized instance ref should be fine. 

maybe update this comment with the specific problem versions as well .

Reply:
which back-compat test fails? Just `N-1` release? 

i was thinking just verifying that the serialized instance ref doesn't contain `EnvFileLoader`